test/components/CivicProfileForms/BasicInfo.test.jsx

```diff
@@ -1,11 +1,84 @@
 import React from 'react';
 import { render } from '@testing-library/react';
-import { expect, it, describe } from 'vitest';
+import { vi, expect, it, describe } from 'vitest';
 import { BasicInfo } from '@components/CivicProfileForms';
+import { useCivicProfile } from '@hooks';
+import userEvent from '@testing-library/user-event';
 
-describe('Basic Info Form', () => {
+vi.mock('@hooks', async () => {
+  const actual = await vi.importActual('@hooks');
+
+  return {
+    ...actual,
+    useCivicProfile: vi.fn()
+  };
+});
+
+describe('Basic info form', () => {
   it('renders', () => {
-    const { getByText } = render(<BasicInfo />);
-    expect(getByText('Basic Info')).not.toBeNull();
+    useCivicProfile.mockReturnValue({ data: {}, isSuccess: true, refetch: vi.fn() });
+    const { getByRole } = render(<BasicInfo />);
+    const legalFirstName = getByRole('textbox', { name: 'Legal first name' });
+    expect(legalFirstName).not.toBeNull();
   });
+
+  it('clears all input fields when you click the clear button', async () => {
+    // const user = userEvent.setup();
+    const mockClear = vi.fn();
+    const basicInfoProfile = {
+      legalFirstName: 'Jane',
+      legalLastName: 'Doe',
+      legalDOB: '1980-12-15',
+      legalGender: 1
+    };
+    useCivicProfile.mockReturnValue({
+      add: mockClear,
+      isSuccess: true,
+      storedDataset: {},
+      refetch: vi.fn()
+    });
+    const { getByRole } = render(<BasicInfo />);
+    const legalFirstName = getByRole('textbox', { name: 'Legal first name' });
+    const legalLastName = getByRole('textbox', { name: 'Legal last name' });
+    const legalDOB = getByRole('textbox', { name: 'Choose date' });
+    const legalGender = getByRole('combobox', { name: 'Gender' });
+    const clearButton = getByRole('button', { name: 'Clear button' });
+
+    await userEvent.type(legalFirstName, basicInfoProfile.legalFirstName);
+    await userEvent.type(legalLastName, basicInfoProfile.legalLastName);
+    await userEvent.type(legalDOB, basicInfoProfile.legalDOB);
+    await userEvent.type(legalGender, `${basicInfoProfile.legalGender}{enter}`);
+
+    await userEvent.click(clearButton);
+
+    expect(legalFirstName.value).toBe('');
+    expect(legalLastName.value).toBe('');
+    expect(legalDOB.value).toBe(null);
```

Comment:
I believe this test is failing because the input field for the date picker is not your typical text field. You can't use the same approach to test the data picker. The same would be said for `legalGender` which is a select field.

If you'd like, we can open up another issue to write unit tests for these inputs.